probe-rs/src/core/mod.rs

```diff
@@ -117,14 +119,75 @@ pub(crate) enum RegisterKind {
 /// Creating a new `RegisterValue` should be done using From or Into.
 /// Converting a value back to a primitive type can be done with either
 /// a match arm or TryInto
-#[derive(Debug, Clone, Copy, PartialEq)]
+#[derive(Debug, Clone, Copy)]
 pub enum RegisterValue {
     /// 32-bit unsigned integer
     U32(u32),
     /// 64-bit unsigned integer
     U64(u64),
 }
 
+impl RegisterValue {
```

Comment:
FYI - as part of adding FP support for ARMv8-a I'm going to need to add a u128 option.  I don't see any issues with merging that with what you have and I'll make sure to sequence this after this PR, but if you know of any place that will fall over a heads up is apprecaited.

Reply:
@thefaxman ... The only thing I can think of is that the debug module uses u64 to store values that are "equivalents" to register values. I think we can find a solution for that, and am happy to tag-team with you on that. Other than that, there are a few `match` arms that will have to be udpated, but that is minor.

Reply:
Thanks - I think this shouldn't be a problem from what I saw during the review.